### ISSUE TITLE
Warnings with ruby-vips 2.1.0

### ISSUE THREAD

**Issue by doits** [janko/image_processing#83]
With `ruby-vips 2.1.0` I get some warning simply requiring `image_processing/vips`:

```ruby
# test.rb
require "image_processing/vips"
```

Output:

```
(test.rb:2846695): VIPS-WARNING **: 14:13:51.612: can't set metadata "scale" on shared image

(test.rb:2846695): VIPS-WARNING **: 14:13:51.612: can't set metadata "offset" on shared image
```

Only happens on ubuntu focal (20.04).

```
libvips42:amd64/focal 8.9.1-2 uptodate

# Gemfile.lock
ruby-vips 2.1.0
image_processing 1.12.1

ruby 2.7.2
```

With `ruby-vips 2.0.17` there is no warning. On my macos development machine there is no warning with `ruby-vips 2.1.0`either (though libvips version is different).

I also have some errors with quality of jpg images after only upgrading to `ruby-vips 2.1.0` (only on my ubuntu server), but I didn't debug it further yet and can't provide steps to reproduce yet – maybe it has something to do with this.

Is there maybe something broken with ruby-vips 2.1.0?

**Comment by janko** [comment]
Thanks for raising an issue. It's a bit difficult to tell what could be the problem, a self-contained example that reproduces the error would definitely helpful.

It's possible that with ruby-vips 2.1.0 adding `#mutate` support there have been some changes which prevent updating metadata on "shared" (non-copied) images.

**Comment by doits** [comment]
Thanks for your response.

You can replicate it simply like this on ubuntu focal:

```bash
# install libvips if not installed
apt install libvips42

# create Gemfile
echo "source 'https://rubygems.org'
gem 'image_processing'" > Gemfile

# install gems
bundle install

# create test file
echo 'require "image_processing/vips"' > test.rb

# run test file
bundle exec ruby test.rb
```

Output for me:

```
(test.rb:85486): VIPS-WARNING **: 23:00:26.254: can't set metadata "scale" on shared image

(test.rb:85486): VIPS-WARNING **: 23:00:26.255: can't set metadata "offset" on shared image
```

Gemfile.lock:

```
GEM
  remote: https://rubygems.org/
  specs:
    ffi (1.15.0)
    image_processing (1.12.1)
      mini_magick (>= 4.9.5, < 5)
      ruby-vips (>= 2.0.17, < 3)
    mini_magick (4.11.0)
    ruby-vips (2.1.0)
      ffi (~> 1.12)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  image_processing

BUNDLED WITH
   2.2.7
```

**Comment by janko** [comment]
Which version of libvips does the `libvips42` package install, or otherwise which version do you have installed? [This page](https://packages.debian.org/stretch/libvips42) to me indicates version 8.4, which is too old, though ImageProcessing has a check where the `require` fails if the libvips version is too old.

Unfortunately I cannot reproduce on my Mac on libvips 8.10, though I could try with an Ubuntu docker image.

**Comment by doits** [comment]
Thanks for testing it. And yes, I can only reproduce it on ubuntu focal. On my mac it works without problems, too. Ubuntu focal has [vips 8.9.1](https://packages.ubuntu.com/focal/libvips42). Maybe it is the combination of `ruby-vips 2.1.0` and `vips 8.9.1`.

**Comment by janko** [comment]
Yeah, if the warning is happening just by *requiring* ruby-vips and checking the libvips version (which is what `image_processing/vips.rb` does), then it seems like it's an issue with ruby-vips & libvips integration itself.

**Comment by doits** [comment]
I tested it again by directly having this in my `test.rb`:

```rb
require "vips"
require "image_processing"

fail "image_processing/vips requires libvips 8.6+" unless Vips.at_least_libvips?(8, 6)

module ImageProcessing
  module Vips
  end
end

# => warnings produced
```

It produces the warning. If I remove the `Vips` module though, there is **no warning**:

```rb
require "vips"
require "image_processing"

fail "image_processing/vips requires libvips 8.6+" unless Vips.at_least_libvips?(8, 6)

module ImageProcessing
end

# => no warnings
```

Not sure what is happening 😕

**Comment by doits** [comment]
Alright, at least I got the line that produces the warnings. It is [sharpen mask](https://github.com/janko/image_processing/blob/master/lib/image_processing/vips.rb#L22-L24):

```rb
require "vips"

SHARPEN_MASK = ::Vips::Image.new_from_array [[-1, -1, -1],
                                             [-1, 32, -1],
                                             [-1, -1, -1]], 24

# => warnings produced
```

Not sure what this means though and who is responsible for the warnings.

**Comment by doits** [comment]
I opened an issue at `ruby-vips`, hopefully they can help/fix it - I think the issue lies there.

**Comment by kokogakayui** [comment]
I had the same problem, that I fixed by upgrading my libvips version to 8.10.6, if that helps.

**Comment by janko** [comment]
Based on the posted information, it seems the issue should be fixed in later libvips versions, so I'm closing this one.
